web-server/src/comment/comment.controller.ts

```diff
@@ -0,0 +1,49 @@
+import {
+    Body,
+    Controller,
+    Get,
+    Query,
+    Post as HttpPost,
+    ValidationPipe,
+    HttpException,
+    HttpStatus
+} from '@nestjs/common';
+import { Repository, TreeRepository } from 'typeorm';
+import { Comment } from './comment.entity';
+import { CommentDto } from './comment.dto';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Post } from '../post/post.entity';
+
+@Controller('comments')
+export class CommentController {
+    constructor(
+        @InjectRepository(Comment) private readonly commentRepository: TreeRepository<Comment>,
+        @InjectRepository(Post) private readonly postRepository: Repository<Post>
+
+    ) { }
+
+    @HttpPost()
+    async create(
+        @Body(
+            new ValidationPipe({
+                whitelist: true,
+                transform: true,
+            }),
+        )
+        data: CommentDto,
+    ): Promise<Comment> {
+        const post: Post = await this.postRepository.findOne(data.postId);
+
+        if (post === undefined) {
```

Comment:
we might want to consider using a not found exception instead of bad request, however Im always unsure which one is the better practice in case of posts that reference other ids => maybe you could find a reference to standard practice?

Reply:
You're right 400 is more about syntax. I guess 422 is a better status code as it is about semantics:  https://tools.ietf.org/html/rfc4918#section-11.2 

The Adyen API uses that response code for example.

Reply:
great